ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -1961,6 +1961,7 @@ public static void updateStats(Statistics stats, long newNumRows,
             newDV = (long) Math.ceil(ratio * oldDV);
           }
           cs.setCountDistint(newDV);
+          cs.setFilterColumn();
```

Comment:
Apparently `public static void updateStats` is called for joins too. Should we limit setting this column to filter operators (we could pass in the context whether this is applicable or not)?

Reply:
I'm not sure - I don't think so...actually because we may end up filtering rows in a join as well - I think it's more accurate to fave this.

I was trying to come up with some query which might be relevant...but they doesn't really made sense...query requirements:
* we need to have a join condition on the FK - where we would not set it as a filterColumn
* we need to have a 2. join condition on the same FK

I think queries like the following make not much sense; and the filter on FK could be movedf to the PK side which has lesser number of rows anyway...so its kinda useless...
```
select 1 from
  torpedos t,
  ships s,
  something ???,
where
  true
-- join#1 condition(x-fk) ; it is required to use t.ship_id
 and t.ship_id = ???
-- the pk-fk condition
 and t.ship_id = s.id
```